projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -313,6 +299,26 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
+    private _navigateToNextParentRow(currentRowEl) {
+         // next is parent summary or next parent row
+         const parent = this.grid.parent;
+         const parentHasSummary = parent.summariesRowList.toArray().length > 0;
+         const parentRowIndex = parseInt(
+             this.getClosestElemByTag(currentRowEl, 'igx-child-grid-row').parentNode.getAttribute('data-rowindex'), 10);
+         const isLastRowInParent = parent.verticalScrollContainer.igxForOf.length - 1 === parentRowIndex;
+         // check if next is sibling
+         const childRowContainer = this.getChildGridRowContainer(this.grid);
+         const nextIsSiblingChild = !!childRowContainer.nextElementSibling;
+         if (isLastRowInParent && parentHasSummary && !nextIsSiblingChild) {
+             // next is parent summary
+             const parentSummary = parent.summariesRowList.toArray()[0].nativeElement;
+             parent.navigation.focusNextRow(parentSummary, 0, this.grid.rootGrid, true);
```

Comment:
you sure the third param should be this.grid.rootGrid? Won't this fail if parent and rootGrid are different?